modules/dists/BlockDist.chpl

```diff
@@ -1641,90 +1494,76 @@ proc BlockArr.doiUseBulkTransferStride(B) {
   // If multi-ddata is possible then we can only do strided bulk
   // transfer when all the blocks have but a single ddata chunk.
   //
+  if this.rank != B.rank then return false;
   return defRectSimpleDData
-         || (oneDData && B._value.oneDData);
+         || (oneDData && chpl__getActualArray(B).oneDData);
 }
 
-proc BlockArr.doiBulkTransfer(B) {
+proc BlockArr.doiBulkTransfer(B, viewDom) {
   if debugBlockDistBulkTransfer then
     writeln("In BlockArr.doiBulkTransfer");
+  const actual = chpl__getActualArray(B);
+  // TODO: how to avoid a deep copy of the domain here? If it's distributed,
+  // it can cause a lot of comms...
+  pragma "no copy" const actDom = chpl__getViewDom(B);
```

Comment:
That's still surprising to me. If it were an *array* it wouldn't be so surprising... but it's returning a domain, and we don't yet have any "domain view"s that would cause it to be a deep copy. Perhaps it's just a compiler TODO.